scene/animation/animation_player.cpp

```diff
@@ -1205,9 +1207,13 @@ void AnimationPlayer::play(const StringName &p_name, float p_custom_blend, float
 	_stop_playing_caches();
 
 	c.current.from = &animation_set[name];
-	c.current.pos = p_from_end ? c.current.from->animation->get_length() : 0;
+
+	if (p_from_end) {
```

Comment:
I'd do this instead:
```cpp
if (c.assigned != name) { // reset
	c.current.pos = p_from_end ? c.current.from->animation->get_length() : 0;
}
```
This was it even allows resuming an animation which was playing backwards and was stop without reset.

Reply:
Agreed, just did it.